src/langchain_google_spanner/vector_store.py

```diff
@@ -832,15 +967,111 @@ def similarity_search_with_score_by_vector(
             List[Document]: List of documents most similar to the query.
         """
 
-        results, column_order_map = self._get_rows_by_similarity_search(
+        results, column_order_map = self._get_rows_by_similarity_search_knn(
             embedding, k, pre_filter
         )
         documents = self._get_documents_from_query_results(
             list(results), column_order_map
         )
         return documents
 
-    def _get_rows_by_similarity_search(
+    def search_by_ANN(
+        self,
+        column_name: str,
+        table_name: str,
+        index_name: str,
+        embedding: List[float],
+        embedding_column_name: str,
+        num_leaves: int,
+        strategy: DistanceStrategy = DistanceStrategy.COSINE,
+        limit: int = None,
+        is_embedding_nullable: bool = False,
+        where_condition: str = None,
+        column_is_nullable: bool = False,
+    ) -> List[Any]:
+        sql = SpannerVectorStore._query_ANN(
+            column_name,
+            table_name,
+            index_name,
+            embedding,
+            embedding_column_name,
+            num_leaves,
+            strategy,
+            limit,
+            is_embedding_nullable,
+            where_condition,
+            column_is_nullable=column_is_nullable,
+        )
+        staleness = self._query_parameters.staleness
+        with self._database.snapshot(
+            **staleness if staleness is not None else {}
+        ) as snapshot:
+            results = snapshot.execute_sql(
+                sql=sql_query,
+            )
+            return list(results)
+
+    @staticmethod
+    def _query_ANN(
+        column_name: str,
+        table_name: str,
+        index_name: str,
+        embedding: List[float],
+        embedding_column_name: str,
+        num_leaves: int,
+        strategy: DistanceStrategy = DistanceStrategy.COSINE,
+        limit: int = None,
```

Comment:
rename it to `K`

Reply:
Won't the K be quite confusing given that that's used inside kNN while this is ANN?